perllib/FixMyStreet/App/Form/Waste/Garden.pm

```diff
@@ -22,9 +22,18 @@ sub details_update_fields {
 
     my $cost_pa = $bin_count == 0 ? 0 : $form->{c}->cobrand->garden_waste_cost_pa($bin_count);
     my $cost_now_admin = $form->{c}->cobrand->garden_waste_new_bin_admin_fee($new_bins);
-    $form->{c}->stash->{cost_pa} = $cost_pa / 100;
-    $form->{c}->stash->{cost_now_admin} = $cost_now_admin / 100;
-    $form->{c}->stash->{cost_now} = ($cost_now_admin + $cost_pa) / 100;
+    if ($data->{apply_discount} && $c->stash->{waste_features}->{ggw_discount_pay_proportion}) {
+        my $proportion_to_pay = $c->stash->{waste_features}->{ggw_discount_pay_proportion};
+        $c->stash->{cost_pa} = (($cost_pa / 100) / 100) * $proportion_to_pay;
+        $c->stash->{per_bin_cost} = ($c->stash->{per_bin_cost} / 100) * $proportion_to_pay;
+        $form->{c}->stash->{cost_now} = ( ( ($cost_now_admin + $cost_pa) / 100 ) * $proportion_to_pay ) / 100;
+        $form->{c}->stash->{cost_now_admin} = (($cost_now_admin / 100) / 100) * $proportion_to_pay;
+    } else {
+        $c->stash->{cost_pa} = $cost_pa / 100;
+        $form->{c}->stash->{cost_now} = ($cost_now_admin + $cost_pa) / 100;
+        $form->{c}->stash->{cost_now_admin} = $cost_now_admin / 100;
+    }
```

Comment:
It feels like this would be clearer if you kept the existing code (what is now in the else), and then if apply_discount/waste_feature is true, apply the discount to all three then - as it is, you're calculating the first part in both places, if you see what I mean.
(Also could consistently use `$c` everywhere, I assume)

Reply:
Also, this all needs doing in Garden/Modify and Garden/Renew as well, I assume - perhaps rather than duplicate in these three places, a cobrand hook to alter the prices in one location?
Renew might be trickier because it sets the prices on the intro page before it's shown, hmm.